runtime/near-vm-errors/src/lib.rs

```diff
@@ -298,6 +297,27 @@ pub enum InconsistentStateError {
     IntegerOverflow,
 }
 
+impl From<FunctionCallError> for FunctionCallErrorSer {
+    fn from(outer_err: FunctionCallError) -> Self {
+        match outer_err {
+            FunctionCallError::CompilationError(e) => FunctionCallErrorSer::CompilationError(e),
+            FunctionCallError::MethodResolveError(e) => FunctionCallErrorSer::MethodResolveError(e),
+            // TODO: consider using FunctionCallErrorSer::HostError
+            FunctionCallError::HostError(ref _e) => {
+                FunctionCallErrorSer::ExecutionError(outer_err.to_string())
+            }
+            // TODO: consider using FunctionCallErrorSer::LinkError
+            FunctionCallError::LinkError { msg } => {
+                FunctionCallErrorSer::ExecutionError(format!("Link Error: {}", msg))
+            }
+            // TODO: consider using FunctionCallErrorSer::WasmTrap
+            FunctionCallError::WasmTrap(ref _e) => {
+                FunctionCallErrorSer::ExecutionError(outer_err.to_string())
+            }
```

Comment:
Yeah, so this is not TODO, but rather intentional design. The "structured" varians for FunctionCallErrorSer are only there to deserialize legacy error, all new data shoudl not depend on details of representation of errors.